### PR TITLE
change MatMulInteger/MatMulInteger16 fallback option

### PR DESCRIPTION
**Pull request by baowenlei**
**Description**: Change fall back option.

**Motivation and Context**
- Change fall back option to handle case when no MKL exist.

### CODE REVIEW

**Comment by tianleiwu** [comment]
Any perf test result for such switch?

File: onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/math/quantize/matmul_integer.cc [hunk 1]
```diff
@@ -139,7 +140,9 @@ static Status EvaluateMatMulInteger(
         //TODO: change to use MLAS when no layout could apply
         tvm::Tensor B_marshalled = tvm_codegen::Transpose(B, {1, 0});
 
-        tvm::Tensor output_tensor = IMatMulExternMKL(A, B_marshalled, output_shape, input_dim, embed_dim, name + "_IMatMulExternMKL");
+        bool use_extern_MKL = (force_mkl || isAVX);
```
Comment (KeDengMS) [inline onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/math/quantize/matmul_integer.cc]: >isAVX [](start = 44, length = 5)

should it be !(isAVX2 or isAVX512)? For example, SSE machines may still fail when trying to run AVX2 code.

**Comment by KeDengMS** [comment]
This is to fix crash when running build without MKL on AVX machines, where AVX2 code path is wrongly used. This change make sure the error becomes more actionable. No performance impact in this change.

---
In reply to: [540087708](https://github.com/microsoft/onnxruntime/pull/2064#issuecomment-540087708) [](ancestors = 540087708)

File: onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/math/quantize/matmul_integer.cc (hunk 1 above)
Comment (baowenlei) [reply onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/math/quantize/matmul_integer.cc]: Right, missing those old machines.

**Pull request merged**